rust_src/src/buffers.rs

```diff
@@ -359,3 +388,40 @@ fn set_buffer(buffer_or_name: LispObject) -> LispObject {
     unsafe { set_buffer_internal(buf.as_mut()) };
     buffer
 }
+
+/// These functions are for debugging overlays.
+
+/// Return a pair of lists giving all the overlays of the current buffer.
+/// The car has all the overlays before the overlay center;
+/// the cdr has all the overlays after the overlay center.
+/// Recentering overlays moves overlays between these lists.
+/// The lists you get are copies, so that changing them has no effect.
+/// However, the overlays you get are the real objects that the buffer uses.
+#[lisp_fn]
+fn overlay_lists() -> LispObject {
+    let cur_buf = ThreadState::current_buffer();
+    fn list_overlays(overlay: Option<LispOverlayRef>) -> LispObject {
+        let mut ol = overlay;
+        let mut ol_list = LispObject::constant_nil();
+        let mut tmp: LispObject;
+        loop {
+            match ol {
+                Some(o) => {
+                    tmp = unsafe {
```

Comment:
Why make this a LispObject to just unwrap it immediately? Same question for the result of `Fcons`.